worldedit-core/src/main/java/com/sk89q/worldedit/EditSession.java

```diff
@@ -3052,7 +3054,12 @@ public int deformRegion(
      */
     public int hollowOutRegion(Region region, int thickness, Pattern pattern, Mask mask) {
         try {
-            final Set<BlockVector3> outside = new LocalBlockVectorSet();
+            final Set<BlockVector3> outside;
+            if (region.getDimensions().getBlockX() > 2048 || region.getDimensions().getBlockZ() > 2048) {
+                outside = new BlockVectorSet();
+            } else {
+                outside = new LocalBlockVectorSet();
+            }
```

Comment:
As this code appears at multiple places now, it would make sense to e.g. have a static factory method in `BlockVector3Set` instead